File: jvm/core/main/scala/com/webank/eggroll/core/containers/meta/StartContainersResponse.scala
package com.webank.eggroll.core.containers.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.meta.Containers

import scala.language.implicitConversions


case class StartContainersResponse(sessionId: String) {
}

object StartContainersResponse {
  implicit def deserialize(byteString: ByteString): StartContainersResponse = {
    val src = Containers.StartContainersResponse.parseFrom(byteString)
    StartContainersResponse(
      sessionId = src.getSessionId
    )
  }

  implicit def serialize(src: StartContainersResponse): Array[Byte] = {
    val builder = Containers.StartContainersResponse.newBuilder()
      .setSessionId(src.sessionId)
    builder.build().toByteArray
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/containers/meta/StartContainersResponse.java
package com.eggroll.core.containers.meta;

import com.eggroll.core.pojo.KillContainersRequest;
import com.eggroll.core.pojo.RpcMessage;
import com.google.protobuf.ByteString;
import com.google.protobuf.InvalidProtocolBufferException;
import com.webank.eggroll.core.meta.Containers;
import lombok.Data;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Data
public class StartContainersResponse implements RpcMessage {
    Logger log = LoggerFactory.getLogger(StartContainersResponse.class);

    private String sessionId;

    public StartContainersResponse() {
    }

    public byte[] serialize() {
        Containers.StartContainersResponse.Builder builder = Containers.StartContainersResponse.newBuilder();
        builder.setSessionId(this.sessionId);
        return builder.build().toByteArray();
    }

    @Override
    public void deserialize(byte[] data) {
        try {
            Containers.StartContainersResponse response = Containers.StartContainersResponse.parseFrom(data);
            this.sessionId = response.getSessionId();
        } catch (InvalidProtocolBufferException e) {
            log.error("StartContainersResponse.deserialize() error :", e);
        }
    }
}